ChallengeUI/src/main/java/com/drivequant/drivekit/challenge/ui/challengelist/viewmodel/ChallengeListViewModel.kt

```diff
@@ -1,76 +1,199 @@
+                } else {
+                    R.string.dk_challenge_no_ranked_challenge_empty_list
+                }
+            }
+
+            ChallengeListCategory.ALL -> R.string.dk_challenge_no_all_challenge
         }
     }
 
-    fun fetchChallengeList(synchronizationType: SynchronizationType = SynchronizationType.DEFAULT) {
-        if (DriveKit.isConfigured()) {
-            DriveKitChallenge.getChallenges(object : ChallengesQueryListener {
-                override fun onResponse(
-                    challengesSyncStatus: ChallengesSyncStatus,
-                    challenges: List<Challenge>) {
-                    val value = when (challengesSyncStatus) {
-                        ChallengesSyncStatus.CACHE_DATA_ONLY,
-                        ChallengesSyncStatus.SUCCESS,
-                        ChallengesSyncStatus.SYNC_ALREADY_IN_PROGRESS -> true
-                        ChallengesSyncStatus.FAILED_TO_SYNC_CHALLENGES_CACHE_ONLY -> false
-                    }
-                    challengeListData = buildChallengeListData(challenges)
-                    syncChallengesError.postValue(value)
+    fun updateSelectedCategory(category: ChallengeListCategory) {
+        if (this.selectedCategory != category) {
+            this.selectedCategory = category
+            update(resettingSelectedDate = true)
+        }
+    }
+
+    @StringRes
+    fun getScreenTagResId() = when (this.selectedCategory) {
+        ChallengeListCategory.ACTIVE -> R.string.dk_tag_challenge_list_active
+        ChallengeListCategory.RANKED -> R.string.dk_tag_challenge_list_ranked
+        ChallengeListCategory.ALL -> R.string.dk_tag_challenge_list_all
+    }
+
+    private fun update(resettingSelectedDate: Boolean = false) {
+        if (resettingSelectedDate) {
+            this.selectedDate = null
+        }
+
+        // Get local challenges for current Tab
+        val challengesByCategory = getChallengesByCategory()
+
+        val dates = computeChallengeYearList(challengesByCategory).toList()
+        if (dates.isNotEmpty()) {
+            val selectedDateIndex: Int = this.selectedDate?.let {
+                val index = dates.indexOf(it)
+                if (index < 0) {
+                    null
+                } else {
+                    index
                 }
-            }, synchronizationType)
+            } ?: (dates.size - 1)
+
+            val date = dates[selectedDateIndex]
+            this.selectedDate = date
+            this.currentChallenges = challengesByCategory.filter { it.startAndEndYear.contains(this.selectedDate) }
+            this.dateSelectorViewModel.configure(dates, selectedDateIndex, DKPeriod.YEAR)
         } else {
-            syncChallengesError.postValue(false)
+            val date = Date().startingFrom(CalendarField.YEAR)
+            this.selectedDate = date
+            this.currentChallenges = challengesByCategory.filter { it.startAndEndYear.contains(this.selectedDate) }
```

Comment:
Dans ce cas, `challengesByCategory` n'est-il pas vide et donc ne peut-on pas écrire directement `this.currentChallenges = emptyList()` ?